qa/integration-tests/src/test/java/io/camunda/it/client/CreateDocumentBatchTest.java

```diff
@@ -0,0 +1,311 @@
+import static org.junit.Assert.assertThrows;
+
+import io.camunda.client.ZeebeClient;
+import io.camunda.qa.util.cluster.TestStandaloneCamunda;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration.TestZeebe;
+import java.io.ByteArrayInputStream;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import org.junit.jupiter.api.Test;
+
+@ZeebeIntegration
+public class CreateDocumentBatchTest {
+
+  private static ZeebeClient zeebeClient;
+
+  @TestZeebe(initMethod = "initTestStandaloneCamunda")
+  private static TestStandaloneCamunda testStandaloneCamunda;
+
+  @SuppressWarnings("unused")
+  static void initTestStandaloneCamunda() {
+    testStandaloneCamunda = new TestStandaloneCamunda();
+  }
+
+  @Test
+  public void shouldCreateDocumentsFromInputStreams() {
+    // given
+    zeebeClient = testStandaloneCamunda.newClientBuilder().build();
+    final var documentContent1 = new ByteArrayInputStream("test one".getBytes());
+    final var documentContent2 = new ByteArrayInputStream("test two".getBytes());
+
+    // when
+    final var response =
+        zeebeClient
+            .newCreateDocumentBatchCommand()
+            .addDocument()
+            .content(documentContent1)
+            .fileName("test1.txt")
+            .done()
+            .addDocument()
+            .content(documentContent2)
+            .fileName("test2.txt")
+            .done()
+            .send()
+            .join();
+
+    // then
+    assertThat(response).isNotNull();
+    assertThat(response.isSuccessful()).isTrue();
+    assertThat(response.getCreatedDocuments()).hasSize(2);
+  }
+
+  @Test
+  public void shouldCreateDocumentsFromDifferentSources() {
+    // given
+    zeebeClient = testStandaloneCamunda.newClientBuilder().build();
+    final var documentContent1 = new ByteArrayInputStream("test one".getBytes());
+    final var documentContent2 = "test two".getBytes();
+
+    // when
+    final var response =
+        zeebeClient
+            .newCreateDocumentBatchCommand()
+            .addDocument()
+            .fileName("test1.txt")
+            .content(documentContent1)
+            .done()
+            .addDocument()
+            .fileName("test2.txt")
+            .content(documentContent2)
+            .done()
+            .send()
+            .join();
+
+    // then
+    assertThat(response).isNotNull();
+    assertThat(response.isSuccessful()).isTrue();
+    assertThat(response.getCreatedDocuments()).hasSize(2);
+  }
+
+  @Test
+  public void shouldCreateFilenameIfMissing() {
+    // given
+    zeebeClient = testStandaloneCamunda.newClientBuilder().build();
+    final var documentContent1 = new ByteArrayInputStream("test one".getBytes());
+
+    // when
+    final var response =
+        zeebeClient
+            .newCreateDocumentBatchCommand()
+            .addDocument()
+            .fileName("file1.txt")
```

Comment:
Filename is supplied here, so is the test is redundant. Can we remove the filename to validate this